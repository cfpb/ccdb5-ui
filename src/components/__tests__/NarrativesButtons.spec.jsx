import ReduxNarrativesButtons, { NarrativesButtons, mapDispatchToProps } from '../RefineBar/NarrativesButtons'
import configureMockStore from 'redux-mock-store'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import React from 'react'
import renderer from 'react-test-renderer'
import thunk from 'redux-thunk'

function setupEnzyme(initialProps={}) {
  const props = Object.assign({
    toggleFlag: jest.fn(),
    options: {
      isChecked: true,
      phase: ''
    }
  }, initialProps)

  const target = mount(<NarrativesButtons {...props} />);

  return {
    props,
    target
  }
}

function setupSnapshot(query={}) {
  const middlewares = [thunk]
  const mockStore = configureMockStore(middlewares)
  const store = mockStore({
    query
  })

  return renderer.create(
    <Provider store={store}>
      <ReduxNarrativesButtons />
    </Provider>
  )
}

describe('initial state', () => {
  it('renders without crashing', () => {
    const target = setupSnapshot()
    const tree = target.toJSON()
    expect(tree).toMatchSnapshot()
  });

  it('pre-check filter based on query', () => {
    const target = setupSnapshot({
      has_narrative: true
    })
    const tree = target.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('checks and disables the filter when searching narratives', () => {
    const target = setupSnapshot({
      searchField: 'complaint_what_happened'
    })
    const tree = target.toJSON()
    expect(tree).toMatchSnapshot()
  })
});

describe('component::NarrativesButtons', () => {
  describe('flag filter toggled', () => {
    it('triggers an update when narratives button is clicked', () => {
      const { target, props } = setupEnzyme()
      const narratives = target.find('#refineToggleNarrativesButton')
      narratives.simulate('click')
      expect(props.toggleFlag).toHaveBeenCalled()
    })

    it('triggers an update when all complaints button is clicked', () => {
      const { target, props } = setupEnzyme()
      const noNarratives = target.find('#refineToggleNoNarrativesButton')
      noNarratives.simulate('click')
      expect(props.toggleFlag).toHaveBeenCalled()
    })
  })

  describe('mapDispatchToProps', () => {
    it('hooks into toggleFlag', () => {
      const dispatch = jest.fn()
      mapDispatchToProps(dispatch).toggleFlag()
      expect(dispatch.mock.calls.length).toEqual(1)
    })
  })
})
