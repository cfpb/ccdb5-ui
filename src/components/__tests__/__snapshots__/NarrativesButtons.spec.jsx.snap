// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial state checks and disables the filter when searching narratives 1`] = `
<section
  className="m-btn-group"
>
  <p>
    Read
  </p>
  <button
    className="a-btn toggle-button selected"
    id="refineToggleNarrativesButton"
    onClick={[Function]}
  >
    Only complaints with narratives
  </button>
  <button
    className="a-btn toggle-button deselected"
    id="refineToggleNoNarrativesButton"
    onClick={[Function]}
  >
    All complaints
  </button>
</section>
`;

exports[`initial state pre-check filter based on query 1`] = `
<section
  className="m-btn-group"
>
  <p>
    Read
  </p>
  <button
    className="a-btn toggle-button selected"
    id="refineToggleNarrativesButton"
    onClick={[Function]}
  >
    Only complaints with narratives
  </button>
  <button
    className="a-btn toggle-button deselected"
    id="refineToggleNoNarrativesButton"
    onClick={[Function]}
  >
    All complaints
  </button>
</section>
`;

exports[`initial state renders without crashing 1`] = `
<section
  className="m-btn-group"
>
  <p>
    Read
  </p>
  <button
    className="a-btn toggle-button deselected"
    id="refineToggleNarrativesButton"
    onClick={[Function]}
  >
    Only complaints with narratives
  </button>
  <button
    className="a-btn toggle-button selected"
    id="refineToggleNoNarrativesButton"
    onClick={[Function]}
  >
    All complaints
  </button>
</section>
`;
